src/miner.erl

```diff
@@ -369,6 +371,35 @@ handle_call({create_block, Metadata, Txns, HBBFTRound}, _From, State) ->
                     end,
                     [],
                     Metadata),
+    %% TODO what do we do about snapshot disagreements?
+    %% IIRC we had agreed to omit
+    Ledger = blockchain:ledger(Chain),
+    {ok, N} = blockchain:config(?num_consensus_members, Ledger),
+    F = ((N - 1) div 3),
+
+    SnapshotHash=
+        case blockchain:config(?snapshot_interval, Ledger) of
+            {ok, Interval} ->
+                case (NewHeight - 1) rem Interval == 0 of
+                    true ->
+                        SHCt =
+                            lists:foldl(
+                              fun({_Idx, #{snapshot_hash := SH}}, Acc) -> % new map vsn
+                                      maps:update_with(SH, fun(V) -> V + 1 end, 1, Acc);
+                                 (_, Acc) ->
+                                      Acc
+                              end,
+                              #{},
+                              Metadata),
+                        case lists:reverse(lists:keysort(2, maps:to_list(SHCt))) of
```

Comment:
So we're sorting the most common ones first?

Reply:
Couple comments here would be good, I think.

Reply:
added